### ISSUE TITLE
Add code docs to new structures for EVM Framework

### ISSUE THREAD

**Issue by rakita** [bluealloy/revm#1882]
https://github.com/bluealloy/revm/pull/1865 have introduced a lot of changes 11k lines of changes. Go over the strictures and comment where they are missing

**Comment by Pana** [comment]
Yes, the documentation is seriously outdated, especially the content in the first chapter.

**Comment by rakita** [comment]
> Yes, the documentation is seriously outdated, especially the content in the first chapter.

Totaly agree
